src/analysis/complex/locally_uniform_limit.lean

```diff
@@ -0,0 +1,167 @@
+    exact div_le_div hM (hf w hw) (sq_pos_of_pos hr) le_rfl },
+  have h2 := circle_integral.norm_integral_le_of_norm_le_const hr.le h1,
+  simp only [cderiv, norm_smul],
+  refine (mul_le_mul le_rfl h2 (norm_nonneg _) (norm_nonneg _)).trans (le_of_eq _),
+  field_simp [_root_.abs_of_nonneg real.pi_pos.le, real.pi_pos.ne.symm, hr.ne.symm],
+  ring
+end
+
+lemma cderiv_sub (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
+  (hf : continuous_on f U) (hg : continuous_on g U) :
+  cderiv r (f - g) z = cderiv r f z - cderiv r g z :=
+begin
+  have h1 : continuous_on (λ (w : ℂ), ((w - z) ^ 2)⁻¹) (sphere z r),
+  { refine ((continuous_id'.sub continuous_const).pow 2).continuous_on.inv₀ (λ w hw h, hr.ne _),
+    rwa [mem_sphere_iff_norm, sq_eq_zero_iff.mp h, norm_zero] at hw },
+  simp_rw [cderiv, ← smul_sub],
+  congr' 1,
+  simp_rw [pi.sub_apply, smul_sub],
+  refine circle_integral.integral_sub _ _,
+  { have := (h1.smul (continuous_on.mono hf ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this },
+  { have := (h1.smul (continuous_on.mono hg ((sphere_subset_closed_ball).trans hzr))),
+    exact continuous_on.circle_integrable hr.le this }
+end
+
+lemma norm_cderiv_sub_lt (hr : 0 < r) (hzr : closed_ball z r ⊆ U)
+  (hfg : ∀ w ∈ sphere z r, ‖f w - g w‖ < M) (hf : continuous_on f U) (hg : continuous_on g U) :
+  ‖cderiv r f z - cderiv r g z‖ < M / r :=
+begin
+  obtain ⟨L, hL1, hL2⟩ : ∃ L < M, ∀ w ∈ sphere z r, ‖f w - g w‖ ≤ L,
+  { have e1 : sphere z r ⊆ U := sphere_subset_closed_ball.trans hzr,
+    have e2 : (sphere z r).nonempty := normed_space.sphere_nonempty.mpr hr.le,
+    have e3 : continuous_on (λ w, ‖f w - g w‖) (sphere z r),
+      from continuous_norm.comp_continuous_on ((hf.mono e1).sub (hg.mono e1)),
+    obtain ⟨x, hx, hx'⟩ := (is_compact_sphere z r).exists_forall_ge e2 e3,
+    exact ⟨‖f x - g x‖, hfg x hx, hx'⟩ },
+  rw [← cderiv_sub hr hzr hf hg],
+  exact (norm_cderiv_le hr hL2).trans_lt ((div_lt_div_right hr).mpr hL1)
+end
+
+lemma tendsto_uniformly_on_cderiv (hf : continuous_on f U) (hδ : 0 < δ)
+  (hFn : ∀ n, continuous_on (F n) U) (hK : cthickening δ K ⊆ U)
+  (hF : tendsto_uniformly_on F f φ (cthickening δ K)) :
+  tendsto_uniformly_on (cderiv δ ∘ F) (cderiv δ f) φ K :=
+begin
+  rw [tendsto_uniformly_on_iff] at hF ⊢,
+  rintro ε hε,
+  filter_upwards [hF (ε * δ) (mul_pos hε hδ)] with n h z hz,
+  simp_rw [dist_eq_norm] at h ⊢,
+  have h2 : ∀ w ∈ sphere z δ, ‖f w - F n w‖ < ε * δ,
+    from λ w hw1, h w (closed_ball_subset_cthickening hz δ (sphere_subset_closed_ball hw1)),
+  convert ← norm_cderiv_sub_lt hδ ((closed_ball_subset_cthickening hz δ).trans hK) h2 hf (hFn n),
+  exact mul_div_cancel _ hδ.ne.symm
+end
+
+end cderiv
+
+section weierstrass
+
+variables [ne_bot φ] (hf : tendsto_locally_uniformly_on F f φ U)
+  (hF : ∀ n, differentiable_on ℂ (F n) U) (hU : is_open U)
+include hf hF hU
+
+lemma tendsto_uniformly_on_deriv_of_cthickening_subset {δ : ℝ} (hδ: 0 < δ) (hK : is_compact K)
```

Comment:
do you really need `[ne_bot φ]` for this one? If the filter is bot, then uniform convergence should always hold, right?